spec/features/nav_menu_spec.rb

```diff
@@ -0,0 +1,77 @@
+
+feature "nav menu" do
+  let(:actor) { create(:admin) }
+
+  around { |ex| with_user_home_subdomain(actor) { ex.run } }
+
+  before do
+    login_as(actor, scope: :user)
+  end
+
+  context "basic" do
+    scenario "index" do
+      # no customizations
+      main_nav_test(setting: "",
+                    have: {"People" => "/users",
+                           "Meals" => "/meals",
+                           "Work" => "/work/signups",
+                           "Reservations" => "/reservations",
+                           "Wiki" => "/wiki"},
+                    not_have: [])
+      # change one link and add another
+      main_nav_test(setting: "[wiki](http://wikipedia.org) [google](http://google.com)",
+                    have: {"People" => "/users",
+                           "Meals" => "/meals",
+                           "Work" => "/work/signups",
+                           "Reservations" => "/reservations",
+                           "Wiki" => "http://wikipedia.org",
+                           "Google" => "http://google.com"},
+                    not_have: [])
+      # add two links
+      main_nav_test(setting: "[Yahoo](http://yahoo.com) [google](http://google.com)",
+                    have: {"People" => "/users",
+                           "Meals" => "/meals",
+                           "Work" => "/work/signups",
+                           "Reservations" => "/reservations",
+                           "Wiki" => "/wiki",
+                           "Yahoo" => "http://yahoo.com",
+                           "Google" => "http://google.com"},
+                    not_have: [])
+      # disable one link
+      main_nav_test(setting: "[wiki](none)",
+                    have: {"People" => "/users",
+                           "Meals" => "/meals",
+                           "Work" => "/work/signups",
+                           "Reservations" => "/reservations"},
+                    not_have: ["wiki"])
+      # no valid data 
+      main_nav_test(setting: "irrelevant gobbledygook",
+                    have: {"People" => "/users",
+                           "Meals" => "/meals",
+                           "Work" => "/work/signups",
+                           "Reservations" => "/reservations",
+                           "Wiki" => "/wiki"},
+                    not_have: [])
+      # complicated url
+      main_nav_test(setting: "[Somewhere out there](http://groucho:swordfish@somewhere.com?foo=bar&baz=frobozz+and+more%21)",
+                    have: {"People" => "/users",
+                           "Meals" => "/meals",
+                           "Work" => "/work/signups",
+                           "Reservations" => "/reservations",
+                           "Wiki" => "/wiki",
+                           "Somewhere out there" => "http://groucho:swordfish@somewhere.com?foo=bar&baz=frobozz+and+more%21"},
+                    not_have: [])
+    end
+  end
+
+  def main_nav_test(setting: "", have: {}, not_have: [])
```

Comment:
I like this supporting method! The setting param can go away now due to the above, but the link checking stuff still makes sense.

It would be nice, though, to test the order. What you could do is Capybara's `find` to get the `ul.nav` element and then run `map` (with a block) on its child nodes to get an array of arrays of form `[text, url]`. Then just change `have` to use that same structure (array instead of hash) and do a direct equality check on them. Then you wouldn't even need `not_have`. This becomes a much stronger test then.

Reply:
Well, that certainly makes for a very terse method!  Keeps it simple....

Reply:
And I've pushed up the new code.